theano/sandbox/cuda/opt.py

```diff
@@ -952,23 +954,23 @@ def local_gpu_reshape(node):
     return False
 
 
-@register_opt()
-@local_optimizer([gpu_from_host, tensor.Flatten])
-def local_gpu_flatten(node):
-    if isinstance(node.op, GpuFromHost):
-        host_input = node.inputs[0]
-        if host_input.owner and \
-           isinstance(host_input.owner.op, tensor.Flatten):
-            outdim = host_input.owner.op.outdim
-            return [GpuFlatten(outdim)(
-                as_cuda_ndarray_variable(host_input.owner.inputs[0]))]
-    if isinstance(node.op, tensor.Flatten):
-        x, = node.inputs
-        outdim = node.op.outdim
-        if x.owner and isinstance(x.owner.op, HostFromGpu):
-            gpu_x, = x.owner.inputs
-            return [host_from_gpu(GpuFlatten(outdim)(gpu_x))]
-    return False
+#@register_opt()
```

Comment:
Oh no, sorry if I wasn't clear. It should not be commented out, but it should transfer ```Flatten``` ops, not ```Reshape```. This is necessary to transfer ```Flatten``` in old pickled theano functions to GPU whenever possible. Optimize ```Flatten```, but use ```gpu_flatten```, not ```GpuFlatten```.